membership-attribute-service/app/models/Attributes.scala

```diff
@@ -19,13 +17,35 @@ object ContentAccess {
   implicit val jsWrite = Json.writes[ContentAccess]
 }
 
+case class CardDetails(last4: String, expirationMonth: Int, expirationYear: Int, forProduct: String) {
+  val asLocalDate = new LocalDate(expirationYear, expirationMonth, 1).plusMonths(1).minusDays(1)
+}
+
+object CardDetails {
+  def fromStripeCard(stripeCard: com.gu.stripe.Stripe.Card, product: String) = {
+    CardDetails(last4 = stripeCard.last4, expirationMonth = stripeCard.exp_month, expirationYear = stripeCard.exp_year, forProduct = product)
+  }
+}
+
+case class Wallet(membershipCard: Option[CardDetails] = None, recurringContributionCard: Option[CardDetails] = None) {
+  val expiredCards: Seq[CardDetails] = Seq(membershipCard, recurringContributionCard).flatten.filter(_.asLocalDate.isBefore(now))
+  // TODO - val cardsExpiringSoon - I assume within 1 calendar month?
+}
+
+object Wallet {
+
+  implicit val cardWriter = Json.writes[CardDetails]
+
+  implicit val jsWrite = Json.writes[Wallet]
+
+}
+
 case class Attributes(
   UserId: String,
   Tier: Option[String] = None,
   MembershipNumber: Option[String] = None,
   AdFree: Option[Boolean] = None,
-  CardExpirationMonth: Option[Int] = None,
-  CardExpirationYear: Option[Int] = None,
+  Wallet: Option[Wallet] = None,
   ContributionPaymentPlan: Option[String] = None,
```

Comment:
In order to be consistent with names, probably we should rename this to `recurringContributionPayment Plan`.